File: PF.Dominio/Model/Construction.cs
using System;
using System.Collections.Generic;

namespace PF.Dominio.Model
{
    public class Construction: Base
    {
        #region Constructor
        public Construction()
        {
            Items = new HashSet<ItemConstruction>();
            Activities = new HashSet<Activity>();
        }
        #endregion

        #region Properties
        public string Name { get; set; }
        public double Cost { get; set; }
        public string Address { get; set; }
        public string Description { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public virtual ICollection<ItemConstruction> Items { get; set; }
        public virtual ICollection<Activity> Activities { get; set; }
        #endregion
    }
}

File: PF.Presentacion/ClientApp/src/app/construction/construction.ts
import { IItemDetalle } from "../calculation/item-detalle";

export interface IConstruction {
    id?: number;
    name?: string;
    cost: number;
    address: string;
    description: string;
    startDate?: Date;
    createdDate: Date;
    modificationDate: Date;
    endDate?: Date;
    itemsDetalle: IItemDetalle[];
}
